src/cdk/text-field/autosize.ts

```diff
@@ -197,11 +197,20 @@ export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
     textarea.style.overflow = 'hidden';
     textarea.placeholder = '';
 
+    const height = textarea.scrollHeight;
+    const {selectionStart, selectionEnd} = textarea;
+
     // Use the scrollHeight to know how large the textarea *would* be if fit its entire value.
-    textarea.style.height = `${textarea.scrollHeight}px`;
+    textarea.style.height = `${height}px`;
     textarea.style.overflow = '';
     textarea.placeholder = placeholderText;
 
+    // On Firefox resizing the textarea will prevent it from scrolling to the caret position.
+    // If the user's cursor is at the end of the text, scroll down to the bottom.
+    if (selectionStart === selectionEnd && selectionEnd === value.length) {
```

Comment:
This only addresses the case where the user is typing at the end of the textarea. Have you tried this:

```ts
const {selectionStart, selectionEnd, selectionDirection} = textarea;

...

textarea.setSelectionStart(selectionStart, selectionEnd, selectionDirection);
```
I *think* this will refocus the textarea at the correct position

Reply:
That's what I was trying initially, but it doesn't seem to work anymore, at least on Firefox.

Reply:
This seems to work for me:

```ts
this._ngZone.runOutsideAngular(() => requestAnimationFrame(() => {
  textarea.setSelectionRange(selectionStart, selectionEnd, selectionDirection);
}));
```

or this:

```ts
this._ngZone.runOutsideAngular(() => requestAnimationFrame(() => {
  textarea.selectionStart = selectionStart;
}));
```

Reply:
Looks like the `requestAnimationFrame` did the trick. Changed.